simpervisor/process.py

```diff
@@ -157,7 +162,7 @@ async def _signal_and_wait(self, signum):
             # We cancel the restart watcher & wait for the process to finish,
             # since we return only after the process has been reaped
             self._restart_process_future.cancel()
-            await self.proc.wait()
+            self.proc.wait()
```

Comment:
Doesn't this need logic like introduced above?

```python
        # Since self.proc is a subprocess.Popen object, wait() method is a blocking
        # call. To prevent the entire program being stuck at this point, we use
        # the poll() method to check whether the child process is alive or not.
        while self.proc.poll() is None:
            await asyncio.sleep(0.1)
```

Reply:
Thanks for catching this. I'll update it in the next commit